core/resources/plugin-java.md

```diff
@@ -1,21 +1,22 @@
 # Default Configuration - Java
 
-The V2 version of the Java Generator.
+The V3 version of the Java Generator.
 
-``` yaml $(java) && $(preview) && !isRequested('@autorest/java')
-# default the v2 generator to using the last stable @microsoft.azure/autorest-core 
-version: ~2.0.4413
+``` yaml $(java) && !$(legacy) && !$(v2) && !isRequested('@microsoft.azure/autorest.java')
+version: ~3.0.6298
 
 use-extension:
-  "@microsoft.azure/autorest.java": "~2.1.88"
+  "@autorest/java": "4.0.4"
 try-require: ./readme.java.md
 ```
 
-``` yaml $(java) && !isRequested('@autorest/java')
+Enable use of the V2 Java generator (and V2 core) with the `--legacy` or `--v2` parameter:
+
+``` yaml $(java) && ($(legacy) || $(v2) || isRequested('@microsoft.azure/autorest.java'))
 # default the v2 generator to using the last stable @microsoft.azure/autorest-core 
 version: ~2.0.4413
 
 use-extension:
-  "@microsoft.azure/autorest.java": "~2.1.88"
+  "@microsoft.azure/autorest.java": "three"
```

Comment:
`three` is that correct?

Reply:
Yep, that's the name of the `npm` tag for v3 of the Java generator.

Reply:
Should we consider using a special npm tag for all the autorest default versions? That way we don't have to ever update these configuration files any longer and it can be controlled by updating that npm tag.

Reply:
We do have V2 and V3 tags but I'm not sure if we're updating them consistently.  I'll see if we can improve on that with some of the automation changes I'll do soon.